File: iOS/v3/CirclePix/Presentation.swift
//
//  Presentation.swift
//  CirclePix
//
//  Created by Mark Burns on 10/28/14.
//  Copyright (c) 2014 Mark Burns. All rights reserved.
//

import Foundation
import CoreData

extension NSManagedObject {
    func setDefaults(p: Presentation) {
        // Set defaults (this is here because awakeFromInsert() is not working)
        p.name = "New Presentation"
        p.guid = NSUUID().UUIDString
        p.lastUpdated = NSDate()
        p.autoplay = true
        p.mediaPropertySite = true
        p.mediaListingVideo = true
        p.mediaQRCodes = true
        p.media24HourInfo = true
        p.mediaShortCode = true
        p.mediaFlyers = true
        p.mediaDvds = true
        p.mediaMobile = true
        p.expRealPortals = true
        p.expPersonalSite = true
        p.expCompanySite = true
        p.expBlogger = true
        p.expYouTube = true
        p.expFacebook = true
        p.expTwitter = true
        p.expCraigslist = true
        p.expLinkedin = true
        p.expPinterest = true
        p.expSeoBoost = true
        p.leadPropertySite = true
        p.leadLeadBee = true
        p.leadFacebook = true
        p.leadMobile = true
        p.leadOpenHouseAnnce = true
        p.commEmail = true
        p.commBatchText = true
        p.commStats = true
        p.displayCompanyLogo = true
        p.displayCompanyName = true
        p.displayAgentImage = true
        p.displayAgentName = true
        p.displayPropAddress = false
        p.displayPropImage = false
        p.desc = ""
        p.listingType = "Residential"
        p.photographyType = "Agent"
        p.music = "None"
        p.narration = "Male"
        p.theme = "CirclePix"
        p.propertyAddress = ""
        p.propertyImage = ""
        p.companyLogo = ""
        p.companyName = ""
        p.agentPhoneNum = ""
        p.agentPhoto = ""
    }
    
    func setDictionaryValues(p: Presentation, data: NSDictionary) {
        p.name = getStringSafe("name", dict: data)
        p.guid = getStringSafe("guid", dict: data)
        p.agentPhoneNum = getStringSafe("agentPhoneNum", dict: data)
        p.agentPhoto = getStringSafe("agentPhoto", dict: data)
        p.autoplay = getBoolSafe("autoplay", dict: data)
        p.commBatchText = getBoolSafe("commBatchText", dict: data)
        p.commEmail = getBoolSafe("commEmail", dict: data)
        p.commStats = getBoolSafe("commStats", dict: data)
        p.companyLogo = getStringSafe("companyLogo", dict: data)
        p.companyName = getStringSafe("companyName", dict: data)
        p.desc = getStringSafe("description", dict: data)
        p.displayAgentImage = getBoolSafe("displayAgentImage", dict: data)
        p.displayAgentName = getBoolSafe("displayAgentName", dict: data)
        p.displayCompanyLogo = getBoolSafe("displayCompanyLogo", dict: data)
        p.displayCompanyName = getBoolSafe("displayCompanyName", dict: data)
        p.displayPropAddress = getBoolSafe("displayPropAddress", dict: data)
        p.displayPropImage = getBoolSafe("displayPropImage", dict: data)
        p.expBlogger = getBoolSafe("expBlogger", dict: data)
        p.expCompanySite = getBoolSafe("expCompanySite", dict: data)
        p.expCraigslist = getBoolSafe("expCraigslist", dict: data)
        p.expFacebook = getBoolSafe("expFacebook", dict: data)
        p.expLinkedin = getBoolSafe("expLinkedin", dict: data)
        p.expPersonalSite = getBoolSafe("expPersonalSite", dict: data)
        p.expPinterest = getBoolSafe("expPinterest", dict: data)
        p.expRealPortals = getBoolSafe("expRealPortals", dict: data)
        p.expSeoBoost = getBoolSafe("expSeoBoost", dict: data)
        p.expTwitter = getBoolSafe("expTwitter", dict: data)
        p.expYouTube = getBoolSafe("expYouTube", dict: data)
        p.leadFacebook = getBoolSafe("leadFacebook", dict: data)
        p.leadLeadBee = getBoolSafe("leadLeadBee", dict: data)
        p.leadMobile = getBoolSafe("leadMobile", dict: data)
        p.leadOpenHouseAnnce = getBoolSafe("leadOpenHouseAnnce", dict: data)
        p.leadPropertySite = getBoolSafe("leadPropertySite", dict: data)
        p.listingType = getStringSafe("listingType", dict: data)
        p.media24HourInfo = getBoolSafe("media24HourInfo", dict: data)
        p.mediaDvds = getBoolSafe("mediaDvds", dict: data)
        p.mediaFlyers = getBoolSafe("mediaFlyers", dict: data)
        p.mediaListingVideo = getBoolSafe("mediaListingVideo", dict: data)
        p.mediaMobile = getBoolSafe("mediaMobile", dict: data)
        p.mediaPropertySite = getBoolSafe("mediaPropertySite", dict: data)
        p.mediaQRCodes = getBoolSafe("mediaQRCodes", dict: data)
        p.mediaShortCode = getBoolSafe("mediaShortCode", dict: data)
        p.music = getStringSafe("music", dict: data)
        p.narration = getStringSafe("narration", dict: data)
        p.photographyType = getStringSafe("photographyType", dict: data)
        p.propertyAddress = getStringSafe("propertyAddress", dict: data)
        p.propertyImage = getStringSafe("propertyImage", dict: data)
        p.theme = getStringSafe("theme", dict: data)
    }
    
    func isValid(dict: NSDictionary) -> Bool {
        let name: NSString? = getValueForKey("name", dict: dict)
        if name == nil {
            return false
        }
        let guid: NSString? = getValueForKey("guid", dict: dict)
        if guid == nil {
            return false
        }
        
        // TODO: Add more validation
        
        return true
    }
    
    func getValueForKey(key: String, dict: NSDictionary) -> NSString? {
        var result: AnyObject? = nil
        result = nullToNil(dict[key])
        
        if result == nil {
            return nil;
        } else {
            let resultStr: NSString? = (result as! NSString?)
            return resultStr
        }
    }
    
    //by KBL 021216
    //Convert NSNull to nil: to avoid error "Could not cast value of type 'nsnull' to 'nsstring' swift"
    func nullToNil(value : AnyObject?) -> AnyObject? {
        if value is NSNull {
            return nil
        } else {
            return value
        }
    }
    
    
    func getStringSafe(key: String, dict: NSDictionary) -> String {
        let nsStr: NSString? = getValueForKey(key, dict: dict)
        if (nsStr == nil) {
            return ""
        } else {
            return nsStr as! String
        }
    }

    func getBoolSafe(key: String, dict: NSDictionary) -> Bool {
        let nsStr: NSString? = getValueForKey(key, dict: dict)
        if (nsStr == nil) {
            return false
        } else {
            return str2Bool(nsStr!)
        }
    }
    
    func str2Bool(str: NSString) -> Bool {
        if str == "true" || str == "1" {
            return true
        }
        return false
    }
    
    func bool2Str(val: Bool) -> String {
        if val == true {
            return "true"
        }
        return "false"
    }
}

class Presentation: NSManagedObject {

    @NSManaged var guid: String
    @NSManaged var lastUpdated: NSDate
    @NSManaged var agentPhoneNum: String
    @NSManaged var agentPhoto: String
    @NSManaged var autoplay: NSNumber
    @NSManaged var commBatchText: NSNumber
    @NSManaged var commEmail: NSNumber
    @NSManaged var commStats: NSNumber
    @NSManaged var companyLogo: String
    @NSManaged var companyName: String
    @NSManaged var desc: String
    @NSManaged var displayAgentImage: NSNumber
    @NSManaged var displayAgentName: NSNumber
    @NSManaged var displayCompanyLogo: NSNumber
    @NSManaged var displayCompanyName: NSNumber
    @NSManaged var displayPropAddress: NSNumber
    @NSManaged var displayPropImage: NSNumber
    @NSManaged var expBlogger: NSNumber
    @NSManaged var expCompanySite: NSNumber
    @NSManaged var expCraigslist: NSNumber
    @NSManaged var expFacebook: NSNumber
    @NSManaged var expLinkedin: NSNumber
    @NSManaged var expPersonalSite: NSNumber
    @NSManaged var expPinterest: NSNumber
    @NSManaged var expRealPortals: NSNumber
    @NSManaged var expSeoBoost: NSNumber
    @NSManaged var expTwitter: NSNumber
    @NSManaged var expYouTube: NSNumber
    @NSManaged var leadFacebook: NSNumber
    @NSManaged var leadLeadBee: NSNumber
    @NSManaged var leadMobile: NSNumber
    @NSManaged var leadOpenHouseAnnce: NSNumber
    @NSManaged var leadPropertySite: NSNumber
    @NSManaged var listingType: String
    @NSManaged var media24HourInfo: NSNumber
    @NSManaged var mediaDvds: NSNumber
    @NSManaged var mediaFlyers: NSNumber
    @NSManaged var mediaListingVideo: NSNumber
    @NSManaged var mediaMobile: NSNumber
    @NSManaged var mediaPropertySite: NSNumber
    @NSManaged var mediaQRCodes: NSNumber
    @NSManaged var mediaShortCode: NSNumber
    @NSManaged var music: String
    @NSManaged var name: String
    @NSManaged var narration: String
    @NSManaged var photographyType: String
    @NSManaged var propertyAddress: String
    @NSManaged var propertyImage: String
    @NSManaged var theme: String

    var isAutoplay: Bool {
        get {
            return Bool(autoplay)
        }
        set {
            autoplay = NSNumber(bool: newValue)
        }
    }
    var isMediaPropertySite: Bool {
        get {
            return Bool(mediaPropertySite)
        }
        set {
            mediaPropertySite = NSNumber(bool: newValue)
        }
    }
    var isMediaPropSite: Bool {
        get {
            return Bool(mediaPropertySite)
        }
        set {
            mediaPropertySite = NSNumber(bool: newValue)
        }
    }
    var isMediaListingVideo: Bool {
        get {
            return Bool(mediaListingVideo)
        }
        set {
            mediaListingVideo = NSNumber(bool: newValue)
        }
    }
    var isMediaQRCodes: Bool {
        get {
            return Bool(mediaQRCodes)
        }
        set {
            mediaQRCodes = NSNumber(bool: newValue)
        }
    }
    var isMedia24HourInfo: Bool {
        get {
            return Bool(media24HourInfo)
        }
        set {
            media24HourInfo = NSNumber(bool: newValue)
        }
    }
    var isMediaShortCode: Bool {
        get {
            return Bool(mediaShortCode)
        }
        set {
            mediaShortCode = NSNumber(bool: newValue)
        }
    }
    var isMediaFlyers: Bool {
        get {
            return Bool(mediaFlyers)
        }
        set {
            mediaFlyers = NSNumber(bool: newValue)
        }
    }
    var isMediaDvds: Bool {
        get {
            return Bool(mediaDvds)
        }
        set {
            mediaDvds = NSNumber(bool: newValue)
        }
    }
    var isMediaMobile: Bool {
        get {
            return Bool(mediaMobile)
        }
        set {
            mediaMobile = NSNumber(bool: newValue)
        }
    }
    
    var isExpRealPortals: Bool {
        get {
            return Bool(expRealPortals)
        }
        set {
            expRealPortals = NSNumber(bool: newValue)
        }
    }
    var isExpPersonalSite: Bool {
        get {
            return Bool(expPersonalSite)
        }
        set {
            expPersonalSite = NSNumber(bool: newValue)
        }
    }
    var isExpCompanySite: Bool {
        get {
            return Bool(expCompanySite)
        }
        set {
            expCompanySite = NSNumber(bool: newValue)
        }
    }
    var isExpBlogger: Bool {
        get {
            return Bool(expBlogger)
        }
        set {
            expBlogger = NSNumber(bool: newValue)
        }
    }
    var isExpYouTube: Bool {
        get {
            return Bool(expYouTube)
        }
        set {
            expYouTube = NSNumber(bool: newValue)
        }
    }
    var isExpFacebook: Bool {
        get {
            return Bool(expFacebook)
        }
        set {
            expFacebook = NSNumber(bool: newValue)
        }
    }
    var isExpTwitter: Bool {
        get {
            return Bool(expTwitter)
        }
        set {
            expTwitter = NSNumber(bool: newValue)
        }
    }
    var isExpCraigslist: Bool {
        get {
            return Bool(expCraigslist)
        }
        set {
            expCraigslist = NSNumber(bool: newValue)
        }
    }
    var isExpLinkedin: Bool {
        get {
            return Bool(expLinkedin)
        }
        set {
            expLinkedin = NSNumber(bool: newValue)
        }
    }
    var isExpPinterest: Bool {
        get {
            return Bool(expPinterest)
        }
        set {
            expPinterest = NSNumber(bool: newValue)
        }
    }
    var isExpSeoBoost: Bool {
        get {
            return Bool(expSeoBoost)
        }
        set {
            expSeoBoost = NSNumber(bool: newValue)
        }
    }
    
    var isLeadPropertySite: Bool {
        get {
            return Bool(leadPropertySite)
        }
        set {
            leadPropertySite = NSNumber(bool: newValue)
        }
    }
    var isLeadLeadBee: Bool {
        get {
            return Bool(leadLeadBee)
        }
        set {
            leadLeadBee = NSNumber(bool: newValue)
        }
    }
    var isLeadFacebook: Bool {
        get {
            return Bool(leadFacebook)
        }
        set {
            leadFacebook = NSNumber(bool: newValue)
        }
    }
    var isLeadMobile: Bool {
        get {
            return Bool(leadMobile)
        }
        set {
            leadMobile = NSNumber(bool: newValue)
        }
    }
    var isLeadOpenHouseAnnce: Bool {
        get {
            return Bool(leadOpenHouseAnnce)
        }
        set {
            leadOpenHouseAnnce = NSNumber(bool: newValue)
        }
    }
    
    var isCommEmail: Bool {
        get {
            return Bool(commEmail)
        }
        set {
            commEmail = NSNumber(bool: newValue)
        }
    }
    var isCommBatchText: Bool {
        get {
            return Bool(commBatchText)
        }
        set {
            commBatchText = NSNumber(bool: newValue)
        }
    }
    var isCommStats: Bool {
        get {
            return Bool(commStats)
        }
        set {
            commStats = NSNumber(bool: newValue)
        }
    }
    
    var isDisplayCompanyLogo: Bool {
        get {
            return Bool(displayCompanyLogo)
        }
        set {
            displayCompanyLogo = NSNumber(bool: newValue)
        }
    }
    var isDisplayCompanyName: Bool {
        get {
            return Bool(displayCompanyName)
        }
        set {
            displayCompanyName = NSNumber(bool: newValue)
        }
    }
    var isDisplayAgentImage: Bool {
        get {
            return Bool(displayAgentImage)
        }
        set {
            displayAgentImage = NSNumber(bool: newValue)
        }
    }
    var isDisplayAgentName: Bool {
        get {
            return Bool(displayAgentName)
        }
        set {
            displayAgentName = NSNumber(bool: newValue)
        }
    }
    var isDisplayPropAddress: Bool {
        get {
            return Bool(displayPropAddress)
        }
        set {
            displayPropAddress = NSNumber(bool: newValue)
        }
    }
    var isDisplayPropImage: Bool {
        get {
            return Bool(displayPropImage)
        }
        set {
            displayPropImage = NSNumber(bool: newValue)
        }
    }

    var jsondata: String {
        get {
            let dataStr: String = ""
            let fields: NSMutableDictionary = NSMutableDictionary()
            
            fields.setValue(guid, forKey: "guid")
            fields.setValue(agentPhoneNum, forKey: "agentPhoneNum")
            fields.setValue(agentPhoto, forKey: "agentPhoto")
            fields.setValue(bool2Str(isAutoplay), forKey: "autoplay")
            fields.setValue(bool2Str(isCommBatchText), forKey: "commBatchText")
            fields.setValue(bool2Str(isCommEmail), forKey: "commEmail")
            fields.setValue(bool2Str(isCommStats), forKey: "commStats")
            fields.setValue(companyLogo, forKey: "companyLogo")
            fields.setValue(companyName, forKey: "companyName")
            fields.setValue(desc, forKey: "description")
            fields.setValue(bool2Str(isDisplayAgentImage), forKey: "displayAgentImage")
            fields.setValue(bool2Str(isDisplayAgentName), forKey: "displayAgentName")
            fields.setValue(bool2Str(isDisplayCompanyLogo), forKey: "displayCompanyLogo")
            fields.setValue(bool2Str(isDisplayCompanyName), forKey: "displayCompanyName")
            fields.setValue(bool2Str(isDisplayPropAddress), forKey: "displayPropAddress")
            fields.setValue(bool2Str(isDisplayPropImage), forKey: "displayPropImage")
            fields.setValue(bool2Str(isExpBlogger), forKey: "expBlogger")
            fields.setValue(bool2Str(isExpCompanySite), forKey: "expCompanySite")
            fields.setValue(bool2Str(isExpCraigslist), forKey: "expCraigslist")
            fields.setValue(bool2Str(isExpFacebook), forKey: "expFacebook")
            fields.setValue(bool2Str(isExpLinkedin), forKey: "expLinkedin")
            fields.setValue(bool2Str(isExpPersonalSite), forKey: "expPersonalSite")
            fields.setValue(bool2Str(isExpPinterest), forKey: "expPinterest")
            fields.setValue(bool2Str(isExpRealPortals), forKey: "expRealPortals")
            fields.setValue(bool2Str(isExpSeoBoost), forKey: "expSeoBoost")
            fields.setValue(bool2Str(isExpTwitter), forKey: "expTwitter")
            fields.setValue(bool2Str(isExpYouTube), forKey: "expYouTube")
            fields.setValue(bool2Str(isLeadFacebook), forKey: "leadFacebook")
            fields.setValue(bool2Str(isLeadLeadBee), forKey: "leadLeadBee")
            fields.setValue(bool2Str(isLeadMobile), forKey: "leadMobile")
            fields.setValue(bool2Str(isLeadOpenHouseAnnce), forKey: "leadOpenHouseAnnce")
            fields.setValue(bool2Str(isLeadPropertySite), forKey: "leadPropertySite")
            fields.setValue(listingType, forKey: "listingType")
            fields.setValue(bool2Str(isMedia24HourInfo), forKey: "media24HourInfo")
            fields.setValue(bool2Str(isMediaDvds), forKey: "mediaDvds")
            fields.setValue(bool2Str(isMediaFlyers), forKey: "mediaFlyers")
            fields.setValue(bool2Str(isMediaListingVideo), forKey: "mediaListingVideo")
            fields.setValue(bool2Str(isMediaMobile), forKey: "mediaMobile")
            fields.setValue(bool2Str(isMediaPropertySite), forKey: "mediaPropertySite")
            fields.setValue(bool2Str(isMediaQRCodes), forKey: "mediaQRCodes")
            fields.setValue(bool2Str(isMediaShortCode), forKey: "mediaShortCode")
            fields.setValue(music, forKey: "music")
            fields.setValue(name, forKey: "name")
            fields.setValue(narration, forKey: "narration")
            fields.setValue(photographyType, forKey: "photographyType")
            fields.setValue(propertyAddress, forKey: "propertyAddress")
            fields.setValue(propertyImage, forKey: "propertyImage")
            fields.setValue(theme, forKey: "theme")

            if NSJSONSerialization.isValidJSONObject(fields) {
                do {
                    let data = try NSJSONSerialization.dataWithJSONObject(fields, options: [])
                    if let str = NSString(data: data, encoding: NSUTF8StringEncoding) {
                        return str as String
                    }
                } catch let error as NSError {
                    print("json error: \(error.localizedDescription)")
                }
            }

            return dataStr
        }
        set {
            // Do nothing
        }
    }
}

File: android/v3/CirclePix/app/src/main/java/com/circlepix/android/data/Presentation.java
package com.circlepix.android.data;

import android.util.Log;

import com.circlepix.android.types.BackgroundMusicType;
import com.circlepix.android.types.ListingType;
import com.circlepix.android.types.NarrationType;
import com.circlepix.android.types.PhotographyType;
import com.circlepix.android.types.ThemeType;

import org.json.JSONException;
import org.json.JSONObject;

import java.lang.reflect.Field;
import java.util.Date;
import java.util.UUID;

public class Presentation {

	private long id;
	private String name = "New Presentation";
	private String description = "";
	private String guid = UUID.randomUUID().toString();
	private Date created = new Date();
	private Date modified = new Date();
	private ListingType listingType = ListingType.residential;
	private boolean autoplay = true;
	private PhotographyType photographyType = PhotographyType.professional;

	private boolean mediaPropertySite = true;
	private boolean mediaListingVideo = true;
	private boolean mediaQRCodes = true;
	private boolean media24HourInfo = true;
	private boolean mediaShortCode = true;
	private boolean mediaFlyers = true;
	private boolean mediaDvds = true;
	private boolean mediaMobile = true;

	private boolean expRealPortals = true;
	private boolean expPersonalSite = true;
	private boolean expCompanySite = true;
	private boolean expBlogger = true;
	private boolean expYouTube = true;
	private boolean expFacebook = true;
	private boolean expTwitter = true;
	private boolean expCraigslist = true;
	private boolean expLinkedin = true;
	private boolean expPinterest = true;
	private boolean expSeoBoost = true;

	private boolean leadPropertySite = true;
	private boolean leadLeadBee = true;
	private boolean leadFacebook = true;
	private boolean leadMobile = true;
	private boolean leadOpenHouseAnnce = true;

	private boolean commEmail = true;
	private boolean commBatchText = true;
	private boolean commStats = true;

	private String companyLogo = "";
	private String companyName = "";
	private String agentPhoto = "";
	private String agentPhoneNum = "";

	private boolean displayCompanyLogo = true;
	private boolean displayCompanyName = true;
	private boolean displayAgentImage = true;
	private boolean displayAgentName = true;
	private boolean displayPropAddress = false;
	private boolean displayPropImage = false;
	private String propertyAddress = "";
	private String propertyImage = "";

	private BackgroundMusicType music = BackgroundMusicType.none;
	private NarrationType narration = NarrationType.male;
	private ThemeType theme = ThemeType.circlepix;

	private boolean selected = false;

	public Presentation() {}

	public Presentation(String name, String desc) {
		this.name = name;
		this.description = desc;
	}


	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getGuid() {
		return guid;
	}

	public void setGuid(String guid) {
		this.guid = guid;
	}

	public ListingType getListingType() {
		return listingType;
	}

	public void setListingType(ListingType listingType) {
		this.listingType = listingType;
	}

	public boolean isDisplayPropImage() {
		return displayPropImage;
	}

	public void setDisplayPropImage(boolean displayPropImage) {
		this.displayPropImage = displayPropImage;
	}

	public String getPropertyAddress() {
		return propertyAddress;
	}

	public void setPropertyAddress(String propertyAddress) {
		this.propertyAddress = propertyAddress;
	}

	public String getPropertyImage() {
		return propertyImage;
	}

	public void setPropertyImage(String propertyImage) {
		this.propertyImage = propertyImage;
	}

	public boolean isAutoplay() {
		return autoplay;
	}

	public void setAutoplay(boolean autoplay) {
		this.autoplay = autoplay;
	}

	public PhotographyType getPhotographyType() {
		return photographyType;
	}

	public void setPhotographyType(PhotographyType photographyType) {
		this.photographyType = photographyType;
	}

	public boolean isMediaPropertySite() {
		return mediaPropertySite;
	}

	public void setMediaPropertySite(boolean mediaPropertySite) {
		this.mediaPropertySite = mediaPropertySite;
	}

	public boolean isMediaListingVideo() {
		return mediaListingVideo;
	}

	public void setMediaListingVideo(boolean mediaListingVideo) {
		this.mediaListingVideo = mediaListingVideo;
	}

	public boolean isMediaQRCodes() {
		return mediaQRCodes;
	}

	public void setMediaQRCodes(boolean mediaQRCodes) {
		this.mediaQRCodes = mediaQRCodes;
	}

	public boolean isMedia24HourInfo() {
		return media24HourInfo;
	}

	public void setMedia24HourInfo(boolean media24HourInfo) {
		this.media24HourInfo = media24HourInfo;
	}

	public boolean isMediaShortCode() {
		return mediaShortCode;
	}

	public void setMediaShortCode(boolean mediaShortCode) {
		this.mediaShortCode = mediaShortCode;
	}

	public boolean isMediaFlyers() {
		return mediaFlyers;
	}

	public void setMediaFlyers(boolean mediaFlyers) {
		this.mediaFlyers = mediaFlyers;
	}

	public boolean isMediaDvds() {
		return mediaDvds;
	}

	public void setMediaDvds(boolean mediaDvds) {
		this.mediaDvds = mediaDvds;
	}

	public boolean isMediaMobile() {
		return mediaMobile;
	}

	public void setMediaMobile(boolean mediaMobile) {
		this.mediaMobile = mediaMobile;
	}

	public boolean isExpRealPortals() {
		return expRealPortals;
	}

	public void setExpRealPortals(boolean expRealPortals) {
		this.expRealPortals = expRealPortals;
	}

	public boolean isExpPersonalSite() {
		return expPersonalSite;
	}

	public void setExpPersonalSite(boolean expPersonalSite) {
		this.expPersonalSite = expPersonalSite;
	}

	public boolean isExpCompanySite() {
		return expCompanySite;
	}

	public void setExpCompanySite(boolean expCompanySite) {
		this.expCompanySite = expCompanySite;
	}

	public boolean isExpBlogger() {
		return expBlogger;
	}

	public void setExpBlogger(boolean expBlogger) {
		this.expBlogger = expBlogger;
	}

	public boolean isExpYouTube() {
		return expYouTube;
	}

	public void setExpYouTube(boolean expYouTube) {
		this.expYouTube = expYouTube;
	}

	public boolean isExpFacebook() {
		return expFacebook;
	}

	public void setExpFacebook(boolean expFacebook) {
		this.expFacebook = expFacebook;
	}

	public boolean isExpTwitter() {
		return expTwitter;
	}

	public void setExpTwitter(boolean expTwitter) {
		this.expTwitter = expTwitter;
	}

	public boolean isExpCraigslist() {
		return expCraigslist;
	}

	public void setExpCraigslist(boolean expCraigslist) {
		this.expCraigslist = expCraigslist;
	}

	public boolean isExpLinkedin() {
		return expLinkedin;
	}

	public void setExpLinkedin(boolean expLinkedin) {
		this.expLinkedin = expLinkedin;
	}

	public boolean isExpPinterest() {
		return expPinterest;
	}

	public void setExpPinterest(boolean expPinterest) {
		this.expPinterest = expPinterest;
	}

	public boolean isExpSeoBoost() {
		return expSeoBoost;
	}

	public void setExpSeoBoost(boolean expSeoBoost) {
		this.expSeoBoost = expSeoBoost;
	}

	public boolean isLeadPropertySite() {
		return leadPropertySite;
	}

	public void setLeadPropertySite(boolean leadPropertySite) {
		this.leadPropertySite = leadPropertySite;
	}

	public boolean isLeadLeadBee() {
		return leadLeadBee;
	}

	public void setLeadLeadBee(boolean leadLeadBee) {
		this.leadLeadBee = leadLeadBee;
	}

	public boolean isLeadFacebook() {
		return leadFacebook;
	}

	public void setLeadFacebook(boolean leadFacebook) {
		this.leadFacebook = leadFacebook;
	}

	public boolean isLeadMobile() {
		return leadMobile;
	}

	public void setLeadMobile(boolean leadMobile) {
		this.leadMobile = leadMobile;
	}

	public boolean isLeadOpenHouseAnnce() {
		return leadOpenHouseAnnce;
	}

	public void setLeadOpenHouseAnnce(boolean leadOpenHouseAnnce) {
		this.leadOpenHouseAnnce = leadOpenHouseAnnce;
	}

	public boolean isCommEmail() {
		return commEmail;
	}

	public void setCommEmail(boolean commEmail) {
		this.commEmail = commEmail;
	}

	public boolean isCommBatchText() {
		return commBatchText;
	}

	public void setCommBatchText(boolean commBatchText) {
		this.commBatchText = commBatchText;
	}

	public boolean isCommStats() {
		return commStats;
	}

	public void setCommStats(boolean commStats) {
		this.commStats = commStats;
	}

	public String getCompanyLogo() {
		return companyLogo;
	}

	public void setCompanyLogo(String companyLogo) {
		this.companyLogo = companyLogo;
	}

	public String getCompanyName() {
		return companyName;
	}

	public void setCompanyName(String companyName) {
		this.companyName = companyName;
	}

	public String getAgentPhoto() {
		return agentPhoto;
	}

	public void setAgentPhoto(String agentPhoto) {
		this.agentPhoto = agentPhoto;
	}

	public String getAgentPhoneNum() {
		return agentPhoneNum;
	}

	public void setAgentPhoneNum(String agentPhoneNum) {
		this.agentPhoneNum = agentPhoneNum;
	}

	public boolean isDisplayCompanyLogo() {
		return displayCompanyLogo;
	}

	public void setDisplayCompanyLogo(boolean displayCompanyLogo) {
		this.displayCompanyLogo = displayCompanyLogo;
	}

	public boolean isDisplayCompanyName() {
		return displayCompanyName;
	}

	public void setDisplayCompanyName(boolean displayCompanyName) {
		this.displayCompanyName = displayCompanyName;
	}

	public boolean isDisplayAgentImage() {
		return displayAgentImage;
	}

	public void setDisplayAgentImage(boolean displayAgentImage) {
		this.displayAgentImage = displayAgentImage;
	}

	public boolean isDisplayAgentName() {
		return displayAgentName;
	}

	public void setDisplayAgentName(boolean displayAgentName) {
		this.displayAgentName = displayAgentName;
	}

	public boolean isDisplayPropAddress() {
		return displayPropAddress;
	}

	public void setDisplayPropAddress(boolean displayPropAddress) {
		this.displayPropAddress = displayPropAddress;
	}

	public BackgroundMusicType getMusic() {
		return music;
	}

	public void setMusic(BackgroundMusicType music) {
		this.music = music;
	}

	public NarrationType getNarration() {
		return narration;
	}

	public void setNarration(NarrationType narration) {
		this.narration = narration;
	}

	public ThemeType getTheme() {
		return theme;
	}

	public void setTheme(ThemeType theme) {
		this.theme = theme;
	}

	public Date getCreated() {
		return created;
	}

	public void setCreated(Date created) {
		this.created = created;
	}

	public Date getModified() {
		return modified;
	}

	public void setModified(Date modified) {
		this.modified = modified;
	}

	public void setJsonData(String jsonString) throws JSONException, IllegalAccessException, IllegalArgumentException {
		// Get fields from json and save them
		JSONObject json = jsonString != null ?
				new JSONObject(jsonString) : new JSONObject();

		Field[] fields = Presentation.class.getDeclaredFields();
		for (Field field : fields) {
			String fieldName = field.getName();
			if (json.has(fieldName)) {
				Object o = json.get(fieldName);
				if (o != null) {
					String val = o.toString();
					if (val.length() > 0) {
						if (field.getType() == String.class) {
							field.set(this, val);
						}
						else if (field.getType() == boolean.class || field.getType() == Boolean.class) {
							field.set(this, Boolean.parseBoolean(val));
						}
						else if (field.getType() == BackgroundMusicType.class) {
							if (BackgroundMusicType.getValueByName(val) != null) {
								field.set(this, BackgroundMusicType.getValueByName(val));
							}
						}
						else if (field.getType() == ListingType.class) {
							if (ListingType.getValueByName(val) != null) {
								field.set(this, ListingType.getValueByName(val));
							}
						}
						else if (field.getType() == NarrationType.class) {
							if (NarrationType.getValueByName(val) != null) {
								field.set(this, NarrationType.getValueByName(val));
							}
						}
						else if (field.getType() == PhotographyType.class) {
							if (PhotographyType.getValueByName(val) != null) {
								field.set(this, PhotographyType.getValueByName(val));
							}
						}
						else if (field.getType() == ThemeType.class) {
							if (ThemeType.getValueByName(val) != null) {
								field.set(this, ThemeType.getValueByName(val));
							}
						}
						else {
							Log.d("Presentation", "Unhandled type: " + field.getType().toString());
						}
					}
				}
			}
		}
	}

	public boolean isSelected() {
		return selected;
	}
	public void setSelected(boolean selected) {
		this.selected = selected;
	}
}
